ts/details.ts

```diff
@@ -11,40 +11,68 @@ import {
   libraryPreparationDefinition,
   receiptDefinition,
 } from "./data/sample";
-import { TableBuilder } from "./component/table-builder";
 import { getSearchParams, updateUrlParams, urls } from "./util/urls";
+import { TabBar } from "./component/tab-bar-builder";
+import { Pair } from "./util/pair";
+import { TableBuilder } from "./component/table-builder";
 
-new TableBuilder(
-  caseDefinition,
-  "casesTableContainer",
-  getSearchParams(),
-  updateUrlParams
-).build();
-
-new TableBuilder(receiptDefinition, "receiptsTableContainer").build();
-
-new TableBuilder(extractionDefinition, "extractionsTableContainer").build();
-
-new TableBuilder(
-  libraryPreparationDefinition,
-  "libraryPreparationsTableContainer"
-).build();
-
-new TableBuilder(
-  getLibraryQualificationsDefinition(urls.rest.libraryQualifications, true),
-  "libraryQualificationsTableContainer"
-).build();
+const tableContainerId = "tableContainer";
 
-new TableBuilder(
-  getFullDepthSequencingsDefinition(urls.rest.fullDepthSequencings, true),
-  "fullDepthSequencingsTableContainer"
-).build();
+// each pair consists of (1) the information required to build the table (definition
+// and title), and (2) its reload function upon selection
+const tables = [
+  new Pair(new Pair(caseDefinition, "Cases"), reload),
```

Comment:
This still has the tab-bar dealing with tables, which it shouldn't. Instead, you can add the table definition as a parameter to the reload function, and these tab definitions could be `Pair<string, () => void`. Here, this would look something like: `new Pair("Cases", () => reload(caseDefinition))`